src/components/common/ColonyMembers/ColonyMembers.tsx

```diff
@@ -1,221 +1,88 @@
-// import { NOT_FOUND_ROUTE } from '~routes/index';
-// import { checkIfNetworkIsAllowed } from '~utils/networks';
-// import { getAllUserRoles } from '~modules/transformers';
-// import { hasRoot, canAdminister } from '~modules/users/checks';
-// import { oneTxMustBeUpgraded } from '~modules/dashboard/checks';
-import { useColonyContext } from '~hooks';
+import React, { useState } from 'react';
+import { defineMessages } from 'react-intl';
+
+import ColonyHomeInfo from '~common/ColonyHome/ColonyHomeInfo';
+import { COLONY_TOTAL_BALANCE_DOMAIN_ID, ROOT_DOMAIN_ID } from '~constants';
+import Members from '~common/Members';
+import NotFoundRoute from '~routes/NotFoundRoute';
+import { useColonyContext, useMobile } from '~hooks';
+import { SpinnerLoader } from '~shared/Preloaders';
+
+import MemberControls from './MemberControls';
+import MembersFilter, {
+  BannedStatus,
+  FormValues,
+  MemberType,
+  VerificationType,
+} from './MembersFilter';
+import TotalReputation from './TotalReputation';
 
 import styles from './ColonyMembers.css';
 
 const displayName = 'common.ColonyMembers';
 
-// const MSG = defineMessages({
-//   editPermissions: {
-//     id: `${displayName}.editPermissions`,
-//     defaultMessage: 'Edit permissions',
-//   },
-//   banAddress: {
-//     id: `${displayName}.banAddress`,
-//     defaultMessage: 'Ban address',
-//   },
-//   unbanAddress: {
-//     id: `${displayName}.unbanAddress`,
-//     defaultMessage: 'Unban address',
-//   },
-// });
+const MSG = defineMessages({
+  loadingText: {
+    id: `${displayName}.loadingText`,
+    defaultMessage: 'Loading Colony',
+  },
+});
 
 const ColonyMembers = () => {
-  // const { extensionId } = useParams<{
-  //   extensionId?: string;
-  // }>();
+  const { colony, loading } = useColonyContext();
 
-  const { canInteractWithColony } = useColonyContext();
+  const [filters, setFilters] = useState<FormValues>({
+    memberType: MemberType.All,
+    verificationType: VerificationType.All,
+    bannedStatus: BannedStatus.All,
+  });
+
+  const [selectedDomainId, setSelectedDomainId] = useState<number>(
+    COLONY_TOTAL_BALANCE_DOMAIN_ID,
+  );
 
-  // if (!isExtensionIdValid) {
-  //   return <NotFoundRoute />;
-  // }
+  const isRootDomain =
+    selectedDomainId === ROOT_DOMAIN_ID ||
+    selectedDomainId === COLONY_TOTAL_BALANCE_DOMAIN_ID;
+
+  const isMobile = useMobile();
+
+  if (loading) {
+    return (
+      <div className={styles.loadingWrapper}>
+        <SpinnerLoader loadingText={MSG.loadingText} />
+      </div>
+    );
+  }
+
+  if (!colony) {
+    return <NotFoundRoute />;
+  }
 
   return (
     <div className={styles.main}>
       <div className={styles.mainContentGrid}>
+        {isMobile && <ColonyHomeInfo showNavigation isMobile />}
         <div className={styles.mainContent}>
-          Colony Members
-          {canInteractWithColony && <p>Colony can be interacted with</p>}
+          {colony && (
```

Comment:
Don't think this needs to be guarded as `Members` is accessing colony context itself. I.e. the logic to ensure colony is defined can live in `Members`